github/Authorization.py

```diff
@@ -136,45 +128,43 @@ def edit(
         :param note_url: string
         :rtype: None
         """
-        assert scopes is github.GithubObject.NotSet or all(
+        assert isinstance(scopes, _NotSetType) or all(
             isinstance(element, str) for element in scopes
         ), scopes
-        assert add_scopes is github.GithubObject.NotSet or all(
+        assert isinstance(add_scopes, _NotSetType) or all(
             isinstance(element, str) for element in add_scopes
         ), add_scopes
-        assert remove_scopes is github.GithubObject.NotSet or all(
+        assert isinstance(remove_scopes, _NotSetType) or all(
             isinstance(element, str) for element in remove_scopes
         ), remove_scopes
-        assert note is github.GithubObject.NotSet or isinstance(note, str), note
-        assert note_url is github.GithubObject.NotSet or isinstance(
-            note_url, str
-        ), note_url
-        post_parameters = dict()
-        if scopes is not github.GithubObject.NotSet:
-            post_parameters["scopes"] = scopes
-        if add_scopes is not github.GithubObject.NotSet:
-            post_parameters["add_scopes"] = add_scopes
-        if remove_scopes is not github.GithubObject.NotSet:
-            post_parameters["remove_scopes"] = remove_scopes
-        if note is not github.GithubObject.NotSet:
-            post_parameters["note"] = note
-        if note_url is not github.GithubObject.NotSet:
-            post_parameters["note_url"] = note_url
+        assert isinstance(note, (_NotSetType, str)), note
+        assert isinstance(note_url, (_NotSetType, str)), note_url
+
+        post_parameters = NotSet.remove_unset_items(
```

Comment:
I like the idea of removing all that `if note is not github.GithubObject.NotSet:` boilerplate code. What if `requestJsonAndCheck` removes all `NotSet` values from `input`, so caller does not have to create a clean `post_parameters`?

Reply:
I don't think so, some caller will need to transform the input. for example, in coms api user can pass `Iterator[T]` for `list[T]`, caller will have to check input anyway. Doesn't make much value to check to in `requestJsonAndCheck`